### PR TITLE
Feature/ncsdk 26629 migration to ncs

### PR DESCRIPTION
**Pull request by robertstypa**
Ref: NCSDK-26629

Signed-off-by: Robert Stypa <robert.stypa@nordicsemi.no>

### CODE REVIEW

**Review by adsz-nordic — APPROVED** [review]

File: ncs/Kconfig [hunk 1]
```diff
@@ -4,6 +4,27 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+# The following nine KConfigs are maintained for backward compatibility with the sdk-nrf-next repository.
+# They are slated for removal once sdk-nrf-next is no longer in use.
+config SOC_NRF54H20_ENGA_CPUAPP
+    bool
+config SOC_NRF54H20_CPUAPP
```
Comment (tomchy) [inline ncs/Kconfig:11]: This symbol should be defined in NCS

File: ncs/Kconfig [hunk 2]
```diff
@@ -4,6 +4,27 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+# The following nine KConfigs are maintained for backward compatibility with the sdk-nrf-next repository.
+# They are slated for removal once sdk-nrf-next is no longer in use.
+config SOC_NRF54H20_ENGA_CPUAPP
```
Comment (tomchy) [inline ncs/Kconfig:9]: Have we ever used this Kconfig?

File: ncs/Kconfig [hunk 3]
```diff
@@ -26,6 +47,7 @@ config SUIT_ENVELOPE_SEQUENCE_NUM
 config SUIT_ENVELOPE_DEFAULT_TEMPLATE
     string "Path to the default envelope template (deprecated)"
     default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/root_with_nordic_top_envelope.yaml.jinja2" if SOC_NRF54H20_CPUAPP
+    default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/root_with_binary_nordic_top.yaml.jinja2" if SOC_NRF54H20_ENGA_CPUAPP
```
Comment (tomchy) [inline ncs/Kconfig:50]: You probably may use `INCLUDE_SECDOM` instead

File: ncs/Kconfig [hunk 4]
```diff
@@ -176,8 +198,14 @@ config SUIT_ENVELOPE_TOP_TEMPLATE
...
 
 config SUIT_ENVELOPE_ROOT_TARGET
-    string "Map root target to custom target to overwite output aftifacts names."
+    string "Map root target to custom target to overwite output artifacts names."
     default "secdom" if SOC_NRF54H20_CPUSEC && !HW_REVISION_SOC1
     default "sysctrl" if SOC_NRF54H20_CPUSYS && !HW_REVISION_SOC1
 
+config SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY
+    string "Path to the folder with nordic top artifacts"
+    default "./"
```
Comment (tomchy) [inline ncs/Kconfig:207]: This symbol may depend on `I'm building in NCS`-style Kconfig

**Review by ahasztag — APPROVED** [review]

File: ncs/Kconfig (hunk 1 above)
Comment (robertstypa) [reply ncs/Kconfig:11]: It seems that CONFIG_SOC_NRF54H20_ENGA_CPUAPP is defined instead.

File: ncs/Kconfig (hunk 2 above)
Comment (robertstypa) [reply ncs/Kconfig:9]: Used  [here](https://github.com/nrfconnect/suit-generator/pull/103/files#diff-4ad152b400bb31725d53b60cfff80cf8ed20105a8d24983374043f55ae906bdfR50) only in NCS context.

File: ncs/Kconfig (hunk 4 above)
Comment (robertstypa) [reply ncs/Kconfig:207]: Good point, will try to improve that part

File: ncs/Kconfig (hunk 3 above)
Comment (robertstypa) [reply ncs/Kconfig:50]: Switching to `INCLUDE_SECDOM` causing some problems while building SDFW and sysctrl projects.
If we would like to use the following condition:
```
default "${ZEPHYR_SUIT_GENERATOR_MODULE_DIR}/ncs/root_with_binary_nordic_top.yaml.jinja2" if !INCLUDE_SECDOM
```

Then condition is met in the SDFW/sysctrl project and wrong template is used.

If we would like to go with `INCLUDE_SECDOM` then we need to define much more complex condition.

File: ncs/Kconfig (hunk 4 above)
Comment (robertstypa) [reply ncs/Kconfig]: What about:
```
config SUIT_ENVELOPE_NORDIC_TOP_DIRECTORY
    string "Path to the folder with nordic top artifacts"
    default "./" if SOC_NRF54H20_ENGA_CPUAPP
    help
      Path to the folder containing nordic-top.suit envelope.
```
`SOC_NRF54H20_ENGA_CPUAPP` is available only in the NCS context.

File: .gitignore [hunk 5]
```diff
@@ -10,3 +10,4 @@
 __pycache__
 /doc/build/*
 /build/*
+suit-generator.log
```
Comment (kszromek-nordic) [inline .gitignore:13]: Will this suffice taking into account rotating character of logfile?

**Review by SylwesterKonczyk — APPROVED** [review]

File: .gitignore (hunk 5 above)
Comment (robertstypa) [reply .gitignore]: Improved!

File: ncs/Kconfig (hunk 2 above)
Comment (robertstypa) [reply ncs/Kconfig:9]: ENGA KConfigs removed since not available anymore in the sdk-nrf

File: ncs/Kconfig (hunk 1 above)
Comment (robertstypa) [reply ncs/Kconfig:9]: Removed since CONFIG_SOC_NRF54H20_ENGA_CPUAPP  not available anymore

File: ncs/Kconfig (hunk 4 above)
Comment (robertstypa) [reply ncs/Kconfig:203]: It seems that we need to create dedicated branch for NCS so there's no need to keep support for both ncs and next.

**Review by tomchy — APPROVED** [review]

**Pull request merged**